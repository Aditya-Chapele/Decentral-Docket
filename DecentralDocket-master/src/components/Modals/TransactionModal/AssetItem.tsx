/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import * as React from "react"
import { Text, View, StyleSheet, Image } from "react-native"
import { TouchableWithoutFeedback } from "react-native-gesture-handler"
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen"
import { textStyles } from "../../../../docs/config"

interface AssetItemProps {
  image: string
  assetsYouHave: any
  totalAssets: any
  name: string
  index: any
  itemTransferFunc: any
  selectedItem: any
  nftId: any
}

const AssetItem = (props: AssetItemProps) => {
  const {
    itemTransferFunc,
    selectedItem,
    nftId,
    image,
    name,
    assetsYouHave,
    totalAssets,
  } = props

  return (
    <TouchableWithoutFeedback onPress={itemTransferFunc}>
      <View
        style={{
          height: hp("8.62%"),
          width: "100%",
          backgroundColor:
            selectedItem.nftId === nftId ? "rgba(190, 190, 181, 1)" : "#F4F5F8",

          justifyContent: "center",
          marginBottom: 10,
          padding: 0,
        }}
      >
        <View
          style={{
            flexDirection: "row",
            width: "100%",
            justifyContent: "space-around",
          }}
        >
          <View
            style={{
              width: wp("100%"),
              flexDirection: "row",
              alignItems: "center",
            }}
          >
            <View
              style={{
                width: wp("24%"),
                // flex: 0.24,
                // marginLeft: wp('13%'),
                alignItems: "center",
                justifyContent: "center",
              }}
            >
              <Image
                style={{ width: "100%", height: "100%" }}
                source={{
                  uri: image,
                }}
              />
            </View>
            <View style={{ width: wp("60%") }}>
              <Text
                style={{
                  fontFamily: textStyles.regularFont,
                  fontSize: hp("2.2%"),
                  color: "#000000",
                  marginLeft: 20,
                  // alignSelf: 'left'
                }}
              >
                {name}
              </Text>
            </View>
          </View>
          <View
            style={{
              // flex: 0.1,
              // width: wp('70%'),
              // backgroundColor: selectedItem.nftId === nftId? '#000' : '#F4F5F8',

              alignItems: "flex-start",
              justifyContent: "center",
              paddingRight: 50,
            }}
          >
            <Text>
              {assetsYouHave}/{totalAssets}
            </Text>
          </View>
        </View>
      </View>
    </TouchableWithoutFeedback>
  )
}

export default AssetItem
