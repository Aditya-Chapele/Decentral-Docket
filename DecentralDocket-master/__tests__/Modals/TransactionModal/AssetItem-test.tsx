import React from "react"
import renderer from "react-test-renderer"
import AssetItem from "../../../src/components/Modals/TransactionModal/AssetItem"

test("renders correctly", () => {
  const tree = renderer
    .create(
      <AssetItem
        assetsYouHave={[]}
        image="image"
        index={0}
        itemTransferFunc={() => console.log("Item transfer function")}
        name="name"
        nftId={"nftid"}
        selectedItem={"selected item"}
        totalAssets={0}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
