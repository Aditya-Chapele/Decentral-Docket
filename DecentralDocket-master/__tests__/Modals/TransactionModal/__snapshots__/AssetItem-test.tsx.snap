// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={{}}
  >
    <View
      style={
        {
          "backgroundColor": "#F4F5F8",
          "height": 115,
          "justifyContent": "center",
          "marginBottom": 10,
          "padding": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "width": 750,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 180,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "image",
                }
              }
              style={
                {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              {
                "width": 450,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000000",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 29.5,
                  "marginLeft": 20,
                }
              }
            >
              name
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "justifyContent": "center",
              "paddingRight": 50,
            }
          }
        >
          <Text>
            /
            0
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;
